File: BackOffice/BackOffice-COCO-TRIP/Datos/Entidades/Actividad.cs
using BackOffice_COCO_TRIP.Datos.Entidades;
using System;

namespace BackOffice_COCO_TRIP.Models
{
  /// <summary>
  /// Clase que contiene los datos asociados a la actividades de los lugares turisticos
  /// </summary>
  public class Actividad: Entidad
  {
    
    private string nombre; //Nombre de la actividad
    private TimeSpan duracion; //Tiempo estimado de duracion de la actividad
    private string descripcion; //Descripcion de la actividad
    private bool activar; //Activar o desactivar actividad

    private Foto foto; //Foto de la actividad

    /// <summary>
    /// Constructor de la clase
    /// </summary>
    public Actividad()
    {
      duracion = new TimeSpan();
      foto = new Foto();
    }

    /// <summary>
    /// Getters y Setters del atributo Nombre
    /// </summary>
    public string Nombre
    {
      get { return nombre; }
      set { nombre = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo Duracion
    /// </summary>
    public TimeSpan Duracion
    {
      get { return duracion; }
      set { duracion = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo Descripcion
    /// </summary>
    public string Descripcion
    {
      get { return descripcion; }
      set { descripcion = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo Activar
    /// </summary>
    public bool Activar
    {
      get { return activar; }
      set { activar = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo Foto
    /// </summary>
    public Foto Foto
    {
      get { return foto; }
      set { foto = value; }
    }

    /// <summary>
    /// Compara si dos objetos de tipo Actividad son iguales
    /// </summary>
    /// <param name="obj">Actividad</param>
    /// <returns>(bool) Si son iguales retorna true</returns>
    public override bool Equals(object obj)
    {
      if (obj != null && obj is Actividad)
      {
        var objeto = obj as Actividad;

        if (!foto.Equals(objeto.foto) || nombre != objeto.nombre
            || duracion.ToString("c") != objeto.duracion.ToString("c") ||
            descripcion != objeto.descripcion || activar != objeto.activar)
        {
          return (false);
        }
        else
        {
          return (true);
        }
      }
      else
      {
        return (false);
      }

    }

    /// <summary>
    /// Sobreescritura recomendada del metodo GetHashCode
    /// </summary>
    /// <returns>ID del objeto</returns>
    public override int GetHashCode()
    {
      return base.GetHashCode();
    }
  }

}

File: FrontOffice/src/dataAccessLayer/domain/actividad.ts
import { Entidad } from './entidad';
import { DateTime } from 'ionic-angular/components/datetime/datetime';
import { Time } from '@angular/common/src/i18n/locale_data_api';
//****************************************************************************************************//
//*****************************************CLASE MENSAJE MODULO 3*************************************//
//****************************************************************************************************//

/**
 * Autores:
 *Jorge Marin
 */

/**
 * Descripcion de la clase:
 * Entidad que contiene los datos de las actividades de los usuarios
 */
export class Actividad extends Entidad
{
    private nombre:string;
    private foto:string;
    private duracion:Time;
    private descripcion:string;
    private activar:boolean
    private idLugar : number;

      /**
     * Retorna el nombre de la actividad
     */
    get Nombre() : string 
    {
        return this.nombre;
    }

    /**
     * Establece el nombre de la actividad
     */
    set Nombre(nombre : string) 
    {
        this.nombre = nombre;
    }

      /**
     * Retorna el nombre de la foto
     */
    get Foto() : string 
    {
        return this.nombre;
    }

    /**
     * Establece el nombre de la foto
     */
    set Foto(foto : string) 
    {
        this.foto = foto;
    }

      /**
     * Retorna la duracion de la actividad
     */
    get Duracion() : Time 
    {
        return this.Duracion;
    }

    /**
     * Establece la duracion de la actividad
     */
    set Duracion(duracion : Time) 
    {
        this.duracion = duracion;
    }

      /**
     * Retorna la descripcion de la actividad
     */
    get Descripcion() : string 
    {
        return this.descripcion;
    }

    /**
     * Establece la descripcion de la actividad
     */
    set Descripcion(descripcion : string) 
    {
        this.descripcion = descripcion;
    }

      /**
     * Retorna el estado  de la actividad
     */
    get Activar() : boolean 
    {
        return this.activar;
    }

    /**
     * Establece el estado de la actividad
     */
    set Actividad(activar : boolean) 
    {
        this.activar = activar;
    }

      /**
     * Retorna el id del lugar
     */
    get Idlugar() : number
    {
        return this.idLugar;
    }

    /**
     * Establece el id del lugar
     */
    set Idlugar(id : number) 
    {
        this.idLugar = id;
    }
}